src/Control/Middleware/TrustedProxyMiddleware.php

```diff
@@ -176,6 +183,22 @@ public function process(HTTPRequest $request, callable $delegate)
         return $delegate($request);
     }
 
+    /**
+     * Make sure the provided value is a valid hostname
+     * @param string $hostname
+     * @return bool|string
+     */
+    private function filterHostname($hostname)
+    {
+        if (defined('FILTER_VALIDATE_DOMAIN') && defined('FILTER_FLAG_HOSTNAME')) {
+            // Only works on PHP 7
+            return filter_var($hostname, FILTER_VALIDATE_DOMAIN, ['flags' => FILTER_FLAG_HOSTNAME]);
+        } else {
+            $regex = "#^(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])#i";
```

Comment:
Do you want to add a unit test covering this regex?

Reply:
Just added a test for all sort of hostnames. It's failing right now. I'll have to confirm what we want to do with `localhost:8080` and punny code.

It's Monday job.